pkg/bundle/internal/target/source.go

```diff
@@ -27,34 +27,62 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/cert-manager/trust-manager/cmd/trust-manager/app/options"
 	trustapi "github.com/cert-manager/trust-manager/pkg/apis/trust/v1alpha1"
-	"github.com/cert-manager/trust-manager/pkg/bundle/internal/target"
+	"github.com/cert-manager/trust-manager/pkg/fspkg"
 	"github.com/cert-manager/trust-manager/pkg/util"
 )
 
-type notFoundError struct{ error }
+type SourceNotFoundError struct{ error }
 
 type selectsNothingError struct{ error }
 
 type invalidSecretSourceError struct{ error }
 
-// bundleData holds the result of a call to buildSourceBundle. It contains the resulting PEM-encoded
+// BundleData holds the result of a call to BuildBundle. It contains the resulting PEM-encoded
 // certificate data from concatenating all the sources together, binary data for any additional formats and
 // any metadata from the sources which needs to be exposed on the Bundle resource's status field.
-type bundleData struct {
-	target.Data
+type BundleData struct {
+	Data
 
-	defaultCAPackageStringID string
+	DefaultCAPackageStringID string
 }
 
-// buildSourceBundle retrieves and concatenates all source bundle data for this Bundle object.
+type BundleBuilder struct {
+	// a cache-backed Kubernetes Client
+	Client client.Client
+
+	// DefaultPackage holds the loaded 'default' certificate package, if one was specified
+	// at startup.
+	DefaultPackage *fspkg.Package
+
+	// Options holds options for the Bundle controller.
+	Options options.Bundle
+}
+
+func (b *BundleBuilder) Init() error {
+	if b.Options.DefaultPackageLocation != "" {
+		pkg, err := fspkg.LoadPackageFromFile(b.Options.DefaultPackageLocation)
+		if err != nil {
+			return fmt.Errorf("must load default package successfully when default package location is set: %w", err)
+		}
+
+		b.DefaultPackage = &pkg
+
+		b.Options.Log.Info("successfully loaded default package from filesystem", "id", pkg.StringID(), "path", b.Options.DefaultPackageLocation)
+	}
+
+	return nil
+}
```

Comment:
suggestion: I think having an `Init` function introduces the risk of an uninitialized `BundleBuilder` struct being used somewhere and blowing things up.

A pattern I really like is to have a `func NewBundleBuilder(client, defaultPackage, options) (*BundleBuilder, error)` - that way, a user can't forget to call `Init` because creating the object and the init checks are done in `NewBundleBuilder`.

(That pattern is more useful when the type is unexported because then it's impossible to create a `bundleBuilder` which hasn't been initialized/validated, but it's not a requirement)

Reply:
> That pattern is more useful when the type is unexported because then it's impossible to create a bundleBuilder which hasn't been initialized/validated

That will require a new interface, right? I am ok with creating a new interface, but are you?